packages/babel-plugin-transform-block-scoping/src/loop.ts

```diff
@@ -254,45 +254,91 @@ export function wrapLoopBody(
     }
   }
   if (varNames.length) {
-    bodyStmts.push(
-      t.variableDeclaration(
-        "var",
-        varNames.map(name => t.variableDeclarator(t.identifier(name))),
-      ),
+    varPath.unshiftContainer(
+      "declarations",
+      varNames.map(name => t.variableDeclarator(t.identifier(name))),
     );
   }
 
-  if (state.breaksContinues.length === 0 && state.returns.length === 0) {
+  const labelNum = state.breaksContinues.length;
+  const returnNum = state.returns.length;
+  if (labelNum + returnNum === 0) {
     bodyStmts.push(t.expressionStatement(call));
+  } else if (labelNum === 1 && returnNum === 0) {
+    for (const path of state.breaksContinues) {
+      const { node } = path;
+      const { type, label } = node;
+      let name = type === "BreakStatement" ? "break" : "continue";
+      if (label) name += " " + label.name;
+      path.replaceWith(
+        t.addComment(
+          t.returnStatement(t.numericLiteral(1)),
+          "leading",
```

Comment:
Nit, but after playing with this in the playground I think it would look better as trailing

Reply:
Yes! I misread something earlier.